### ISSUE TITLE
"failed apply web serve" when trying to serve current directory.

### ISSUE THREAD

**Issue by jkseppan** [tailscale/tailscale#10213]
### What is the issue?

I tried `tailscale serve .` and `tailscale serve ./` and both output the following:
```
❯ ./bin/tailscale serve ./
error: failed apply web serve: only localhost or 127.0.0.1 proxies are currently supported

try `tailscale serve --help` for usage info
```
This is misleading because this does in fact support serving directories, it just needs to be a full path:
```
❯ ./bin/tailscale serve "$(pwd)"
Available within your tailnet:
...
```
Perhaps the error message could say that in addition to localhost proxies, full directory paths are supported.
(Even better if `.` or `./` would work but I can totally see how that would clash with parsing other things one could serve.)

I'm running the macOS open source version

```
❯ ./bin/tailscale --version
1.52.0-ERR-BuildInfo
  go version: go1.21.4
```

### Steps to reproduce

```
go install tailscale.com/cmd/tailscale{,d}@v1.52.0
sudo ~/go/bin/tailscaled install-system-daemon
~/go/bin/tailscale up
~/go/bin/tailscale serve ./
```

### OS

macOS

### OS version

macOS Sonoma 14.1.1

### Tailscale version

1.52.0

### Bug report

BUG-1bea012618b4436029e664868048d575b1a20e7b3828c58174e5a5a7aafcea71-20231112073238Z-5e86631352e96d98

**Comment by EDR-Hat** [comment]
Was just testing out funnel and got a similar error message on Linux (Manjaro 6.1.84-1)

It still functions as long as I put full paths.

**Comment by clarkezone** [comment]
i also hit this today, would be nice if there was at least an intelligable error message if relative paths aren't supported which they appear not to be

**Comment by jonocodes** [comment]
Same here
> tailscale version
1.58.2
  go version: go1.21.9
> uname -av
Linux dobro 6.1.88 #1-NixOS SMP PREEMPT_DYNAMIC Sat Apr 27 15:07:18 UTC 2024 x86_64 GNU/Linux

**Comment by EDIflyer** [comment]
Yep - I was confused by this as after watching the video at https://www.youtube.com/watch?v=MpxmfpCl20c I hadn't appreciated the full directory was required.  It would definitely be good if it could add a message to the user to say the full path is required (or even better cope with relative paths!)

**Comment by corneliusroemer** [comment]
Thanks to this issue, I figured it out! This misleading error message should really be fixed. I've made a Stackoverflow Q&A to make it easier to find the solution: https://stackoverflow.com/questions/79420723/when-trying-to-use-tailscale-serve-getting-error-failed-apply-web-serve-onl/79420724#79420724

**Comment by MarshCastle** [comment]
+1
